monkey/monkey_island/cc/ui/src/components/utils/ServerUtils.tsx

```diff
@@ -17,8 +19,15 @@ export function didAllAgentsShutdown() {
   })
 }
 
+export function getCollectionObject(collectionEndpoint: APIEndpoit, key: string) {
+  return IslandHttpClient.get(collectionEndpoint)
+      .then(res => {
+        return res.body.reduce((prev, curr) => ({...prev, [curr[key]]: curr}), {});
```

Comment:
Took some digging on my part to realize that this takes an array of items, and converts it into a mapping of those items, keyed by the `key` provided as an argument. I know it's small, but it may be useful to split this out into its own function and give it a useful name such as `toMapIndexedBy(key)`